src/python/WMComponent/AgentStatusWatcher/AgentStatusPoller.py

```diff
@@ -255,29 +257,44 @@ def collectWMBSInfo(self):
 
         return results
 
-    def checkProxyLifetime(self, agInfo):
+    def checkCredLifetime(self, agInfo, cred_type):
         """
-        Check the proxy lifetime (usually X509_USER_CERT) and raise either
-        a warning or an error if the proxy validity is about to expire.
+        Check the credential lifetime. Usually X509_USER_PROXY or X509_USER_CERT
+        and raise either a warning or an error if the proxy validity is about to expire.
         :param agInfo: dictionary with plenty of agent monitoring information in place.
+        :param cred_type: credential type, can be: "proxy" or "certificate"
         :return: same dictionary object plus additional keys/values if needed.
         """
-        secsLeft = self.proxy.getTimeLeft(proxy=self.proxyFile)
-        logging.debug("Proxy '%s' lifetime is %d secs", self.proxyFile, secsLeft)
+        thresholds = {'proxy': {'error': 3, 'warning': 5},
```

Comment:
How about actually defining it in the __init__ method as an instance variable `self.proxyThres = ...`? Second line has a too many leading space.

Reply:
Done.